core/block/editor/stext/text.go

```diff
@@ -274,10 +274,19 @@ func (t *textImpl) newSetTextState(blockId string, ctx session.Context) *state.S
 
 func (t *textImpl) flushSetTextState(_ smartblock.ApplyInfo) error {
 	if t.lastSetTextState != nil {
+		applyFlags := []smartblock.ApplyFlag{smartblock.NoHooks}
+		textChanged, err := t.isLastTextBlockChanged()
```

Comment:
as I mentioned in PM, you should encapsulate working with KeepInternalFlags as much as possible, like

```
applyFlags = []smatblock.ApplyFlag{smartblock.NoHooks}
if t.shouldKeepInternalFlags() {
   ...
}
```

take care to name those functions to be easily searchable via `Find in files`

Reply:
Added `shouldKeepInternalFlags` function in textImpl and basic